scripts/west_commands/runners/nrfjprog.py

```diff
@@ -19,6 +19,38 @@
 def has_region(region_filter, hex_file):
     return any(region_filter(addr) for addr in IntelHex(hex_file).addresses())
 
+# Helper function for nRF53
+def is_cpunet(addr):
+    return 0x01000000 <= addr < 0x10000000
+
+# Helper function for nRF53
+def has_cpunet_region(hex_file):
+    return has_region(is_cpunet, hex_file)
+
+# Helper function for nRF53
+def has_cpuapp_region(hex_file):
+    return has_region(lambda addr: not is_cpunet(addr), hex_file)
+
+# Helper function for nRF53
+# Generates 2 new hex files where cpunet flash contents are in one,
+# and all other flash contents are in the other.
+def split_hex_file(hex_file):
+    hex_file_dir = os.path.dirname(hex_file)
+    cpuapp_hex = os.path.join(hex_file_dir, "nrf53_cpuapp.hex")
```

Comment:
I am definitely OK with this since it fixes a bug, but I'm not sure if we should be doing this in the build system instead longer-term to avoid potentially expensive hex file recreation even when dependencies are up to date.

@tejlmand  what do you think about this in the longer term?

Reply:
I tested it now, `split_hex_file()` takes `0.4274630546569824` seconds for my example. I added hashing and a depfile, and brought it down to `0.0014848709106445312`. I'll add it as a separate commit.

Reply:
to be honest, we have all the information we need already, so we don't need to first merge, and then split the hex file again.

I would prefer we make a more generic approach so the the knowledge of multiple hex files can be passed to other parts of the system.

Reply:
also note, the proposed solution here only works for `nrfjprog` runner, and not openocd.